### ISSUE TITLE
The printf family of functions doesn't support the S conversion specifier

### ISSUE THREAD

**Issue by GabrielRavier** [jart/cosmopolitan#771]
Support for using `%S` with the printf family of functions is currently missing.

It should be simple enough to implement as an alias to `%ls`.

If you want it, I've written a patch to do so (though since I am not particularly familiar with this code,  I may have missed an easier way of implementing it):

```patch
From f50dfc64d100fb72df4544c0b95a33bb86cadc0c Mon Sep 17 00:00:00 2001
From: Gabriel Ravier <gabravier@gmail.com>
Date: Thu, 23 Mar 2023 17:16:19 +0100
Subject: [PATCH] Implement S conversion specifier for printf-related functions

POSIX specifies that a conversion specifier of S must be interpreted
the same way as %ls. This patch implements this.
---
 libc/fmt/fmt.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/libc/fmt/fmt.c b/libc/fmt/fmt.c
index d21133f53..99b8e8afd 100644
--- a/libc/fmt/fmt.c
+++ b/libc/fmt/fmt.c
@@ -359,10 +359,15 @@ _Hide int __fmt(void *fn, void *arg, const char *format, va_list va) {
           p = "\r\e[K";
           goto FormatString;
         }
+      case 'S':
+        // Specified by POSIX to be equivalent to %ls
+        signbit = 63;
+        goto FormatStringPNotFetchedYet;
       case 'q':
         flags |= FLAGS_QUOTE;
         // fallthrough
       case 's':
+      FormatStringPNotFetchedYet:
         p = va_arg(va, void *);
       FormatString:
         if (__fmt_stoa(out, arg, p, flags, prec, width, signbit, qchar) == -1) {
-- 
2.39.2
```

**Comment by G4Vi** [comment]
@GabrielRavier Thank you for the reports and patches, the printf family functions definitely have a lot of room for improvement.

Before we can accept your patches, We need you to assign your copyright to Justine Tunney (the creator of the project), see https://github.com/jart/cosmopolitan/blob/master/CONTRIBUTING.md#copyright-assignment for details and reasoning.

Pull requests are welcome and preferred, but I can also review and merge patches. Also adding tests is highly encouraged, for fmt many of the tests are here https://github.com/jart/cosmopolitan/blob/master/test/libc/fmt/fmt_test.c

**Comment by GabrielRavier** [comment]
I've done the copyright assignment already (or at least my Sent Mail folder on the email tied to my GitHub account indicates to me that I've sent an email to `jtunney@gmail.com` with what seems to me like an appropriate copyright assignment as specified in `CONTRIBUTING.md`), but so I think I'll go and make PRs for these issues to make things simpler

**Comment by G4Vi** [comment]
Fixed by #786
